### PR TITLE
Use Development environment for Sync by default in Debug builds

### PR DESCRIPTION
**Pull request by ayoy**
Task/Issue URL: https://app.asana.com/0/414235014887631/1205489036222324/f

**Description**:
When running the app from Xcode, use Development environment for Sync by default.

**Steps to test this PR**:
1. Run the app from Xcode
2. Identify as internal user
4. Sign up for Sync (Settings -> Sync)
5. Verify that requests are being sent to Development environment (dev-sync-use.duckduckgo.com).

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/AppDelegate.swift [hunk 1]
```diff
@@ -191,12 +191,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+#if DEBUG
+        let defaultEnvironment = ServerEnvironment.development
+#else
+        let defaultEnvironment = ServerEnvironment.production
+#endif
+
         let environment = ServerEnvironment(
             UserDefaultsWrapper(
                 key: .syncEnvironment,
                 defaultValue: ServerEnvironment.production.description
```
Comment (brindy) [inline DuckDuckGo/AppDelegate.swift]: This results in the default still being production. Suggested fix:

```suggestion
                defaultValue: defaultEnvironment.description
```

**Review by brindy — CHANGES_REQUESTED** [review]
Not quite right.  Have suggested a fix.  Same will need to be applied to macOS too.

Screenshot 2023-09-14 at 16 34 55

**Comment by ayoy** [comment]
Thanks for spotting it @brindy. Have pushed a fix to both repos!

**Review by brindy — APPROVED** [review]
LGTM!

**Pull request merged**